lib/api/completeMultipartUpload.js

```diff
@@ -153,235 +154,79 @@ function completeMultipartUpload(authInfo, request, log, callback) {
+            return services.getMPUparts(mpuBucket.getName(), uploadId, log,
+                (err, result) => {
                     if (err) {
                         return next(err, destBucket);
                     }
-                    return next(null, destBucket, objMD, mpuBucket,
-                    jsonList, storedMetadata, completeObjData, mpuOverviewKey);
+                    const storedParts = result.Contents;
+                    return next(null, destBucket, objMD, mpuBucket, storedParts,
+                        jsonList, storedMetadata, location, mpuOverviewKey);
                 });
-            }
-            return next(null, destBucket, objMD, mpuBucket, jsonList,
-            storedMetadata, null, mpuOverviewKey);
         },
-        function retrieveParts(destBucket, objMD, mpuBucket, jsonList,
-        storedMetadata, completeObjData, mpuOverviewKey, next) {
-            // means MPU completion was done on third-party backend,
-            // so part info is not in our metadata and related steps
-            // should be skipped
-            if (completeObjData) {
-                return next(null, destBucket, objMD, mpuBucket, null,
-                jsonList, storedMetadata, completeObjData, mpuOverviewKey);
-            }
-            return services.getMPUparts(mpuBucket.getName(), uploadId, log,
-                (err, result) => {
+        function ifMultipleBackend(destBucket, objMD, mpuBucket, storedParts,
+        jsonList, storedMetadata, location, mpuOverviewKey, next) {
+            if (config.backends.data === 'multiple') {
+                return multipleBackendGateway.completeMPU(objectKey,
+                uploadId, location, jsonList, storedParts, bucketName, log,
+                (err, completeObjData) => {
                     if (err) {
                         return next(err, destBucket);
                     }
-                    const storedParts = result.Contents;
-                    return next(null, destBucket, objMD, mpuBucket,
-                        storedParts, jsonList, storedMetadata, null,
+                    return next(null, destBucket, objMD, mpuBucket, storedParts,
+                        jsonList, storedMetadata, completeObjData,
                         mpuOverviewKey);
                 });
+            }
+            return next(null, destBucket, objMD, mpuBucket, storedParts,
+                jsonList, storedMetadata, null, mpuOverviewKey);
         },
         function processParts(destBucket, objMD, mpuBucket, storedParts,
             jsonList, storedMetadata, completeObjData, mpuOverviewKey, next) {
-            let calculatedSize = 0;
-            if (completeObjData) {
+            // means MPU completion was done on third-party backend,
+            // and Etag was returned, so part info is not in our metadata
+            //  and related steps should be skipped
+            if (completeObjData && !completeObjData.calculateEtag) {
```

Comment:
Your comment `Etag was returned` and then you check `!completeObjData.calculateEtag`. AWS should not be completeObjData.calculateEtag equals true?